src/rules/pr.ts

```diff
@@ -31,4 +31,20 @@ export let pr = {
     }
   },
 
+  /** Warns if http:// was used instead of https:// */
+  async http() {
+    const files = ([] as string[]).concat(
+      danger.git.created_files  || [],
+      danger.git.modified_files || [],
+    );
+
+    for (const file of files) {
+      const diff = await danger.git.diffForFile(file);
+
+      if (diff && diff.added.match(/http:/ig)) {
+        warn('Detected insecure: use of http instead of https on file ' + file);
```

Comment:
no need to warn on every file, this is an approach that lead danger to be very verbose and mostly on our past implementation.

Also, breaking when you found an issue will also make it faster, as you don't have to keep opening/reading every file for this PR if you already know that this problem exists.

Could you check my `console.log` implementation and let me know what you think? https://github.com/indigotech/dangerjs-plugin/blob/bugfix/main/src/rules/node.ts#L61-L63

Reply:
Yeah, I thought of adding this feature with the same behaviour of the old danger but a I totally agree with the verbose point. I Will break like the `console.log` implementation

Reply:
@felipesabino @alanraso 
My fear is that by breaking the checks once we find an issue we end up making the process as a whole more bureaucratic. Consider the case when de developer writes code that triggers 3 different errors. Instead of showing at once where are all the issues. We simply show her one error. Once she fixes it, we show another, and then another...

IMO this back and forth would make the process even more frustrating...

Reply:
@melloflavio  Well, it's not an error, it's just a warning to let you know that your PR needs extra attention.

But I got your point and TBH I am not sure which one is the best approach...

We have a culture problem that people just don't write what they are doing nor read when automatic messages trying to inform what they should be doing. I guess this is the first issue we should tackle and IMO reducing the number of false positives and repeated text might help ("might", I am guessing here)

If Danger says in the PR "there is a string X in your code, you might want to remove/replace it" and the dev can't even cmd+f "x" on the PR to remove all of it, then I am not confident that  the issue is with Danger or (lack of?) warning texts